pkg/endpoint/policy.go

```diff
@@ -913,6 +983,11 @@ func (e *Endpoint) SetIdentity(owner Owner, identity *identityPkg.Identity) {
 
 	e.runIdentityToK8sPodSync()
 
+	// Whenever the identity is updated, propagate change to key-value store
```

Comment:
If the identity was updated, when/how are you deleting the old identity from the kvstore?

Reply:
I'm updating the same key every time (IPv4 and IPv6) whenever the endpoint's identity is updated. Thus, the old identity will be removed upon update because it's for the same key.

Reply:
Oh, the key is `/IP/secID` I thought it was `secID/IP`. So when are you deleting the old identities from `identityToIPCache`?

Reply:
The cache is updated when the watcher receives an event. The unit tests I have added should cover this.

Reply:
Fixed after offline discussion.